### DISCUSSION TITLE
Search in `fastn`

### DISCUSSION THREAD

**Discussion by gargajit** [fastn-stack/fastn#626] [Ideas & RFCs]
- [x] [Search using sitemap processor](https://fastn.com/blog/search/)
- [ ] Local index based search
- [ ] Algolia or other remote index based search

------
To enhance the user experience and improve accessibility to documentation, could we consider implementing a search bar in the `fastn` library documentation, similar to the search bar in the `tailwindCSS` documentation

Asked by @saurabh-lohiya
(Rephrased)

![image](https://user-images.githubusercontent.com/118595104/223373291-6b25addf-1c2c-44f4-bf10-d594c09b7747.png)

**Comment by amitu** [comment]
Search is on our roadmap, it is blocked on some features we have to build first. Some sort of indexer we want, which re-indexes any package on every change. We have to worry about where to store the index, and how to query the index when user searches. We also have to worry about how to make this work in static hosting situations.

**Comment by amitu** [comment]
The initial version of search is live.

**Comment by amitu** [comment]
https://github.com/nextapps-de/flexsearch
